src/static/js/Changeset.js

```diff
@@ -968,18 +976,24 @@ class TextLinesMutator {
         this._curLine += newLines.length;
         // insert the remaining chars from the "old" line (e.g. the line we were in
         // when we started to insert new lines)
-        this._curSplice.push(theLine.substring(lineCol));
-        this._curCol = 0; // TODO(doc) why is this not set to the length of last line?
+        const remaining = theLine.substring(lineCol);
+        if (remaining !== '') this._curSplice.push(remaining);
```

Comment:
If `remaining === ''`, then that means that `this._curCol` points just past the line's newline, correct? If so, I think the better fix would be to set `this._curCol` to 0 and increment `this._curLine` before doing anything else.

Also, `this._curCol` pointing just past the newline feels like a violation of an invariant. Maybe instead of checking if `remaining !== ''`, it should assert that and we should fix whatever logic is causing `this._curCol` to point past the newline.

Reply:
There is not necessarly a newline at every line. Etherpad uses it with lines that always contain newlines (e.g. rep.lines), except for `mutateAttributionLines`, but this has other logic.
Also take a look at the test case, this is valid imo.

Reply:
I do understand what you wrote in the first paragraph, though. If there is a newline, we would never go "beyond" that. Let me test